Messerli.ChangeCase/StringCaseExtensions.SplitResult.cs

```diff
@@ -11,13 +11,13 @@ public static partial class StringCaseExtensions
             public readonly int NextStartIndex;
 
             public SplitResult(int nextStartIndex, Option<string> result = default)
-            {
-                Result = result
+                => (Result, NextStartIndex) = (FlattenResult(result), nextStartIndex);
+
+            private static string FlattenResult(Option<string> result)
```

Comment:
Unrelated nitpick (feel free to resolve): `result` is only none when used as the seed for a `Sequence.Generate`, right? Wouldn't it make sense to just pass in `string.Empty` from there and make `result` non-optional?

Reply:
Addressed in 0d227abcfa9d64e525f701a020b8153b9e0df991